metadata-ingestion/src/datahub/ingestion/source/cassandra/cassandra_utils.py

```diff
@@ -0,0 +1,172 @@
+        "counter": NumberTypeClass,
+        "decimal": NumberTypeClass,
+        "double": NumberTypeClass,
+        "float": NumberTypeClass,
+        "int": NumberTypeClass,
+        "smallint": NumberTypeClass,
+        "tinyint": NumberTypeClass,
+        "varint": NumberTypeClass,
+        # Dates
+        "date": DateTypeClass,
+        # Times
+        "duration": TimeTypeClass,
+        "time": TimeTypeClass,
+        "timestamp": TimeTypeClass,
+        # Strings
+        "text": StringTypeClass,
+        "ascii": StringTypeClass,
+        "inet": StringTypeClass,
+        "timeuuid": StringTypeClass,
+        "uuid": StringTypeClass,
+        "varchar": StringTypeClass,
+        # Records
+        "geo_point": RecordTypeClass,
+        # Arrays
+        "histogram": ArrayTypeClass,
+    }
+
+    @staticmethod
+    def get_column_type(cassandra_column_type: str) -> SchemaFieldDataType:
+        type_class: Optional[
+            Type
+        ] = CassandraToSchemaFieldConverter._field_type_to_schema_field_type.get(
+            cassandra_column_type
+        )
+        if type_class is None:
+            logger.warning(
+                f"Cannot map {cassandra_column_type!r} to SchemaFieldDataType, using NullTypeClass."
+            )
+            type_class = NullTypeClass
+
+        return SchemaFieldDataType(type=type_class())
+
+    def __init__(self) -> None:
+        self._prefix_name_stack: List[str] = [self.version_string]
+
+    def _get_cur_field_path(self) -> str:
+        return ".".join(self._prefix_name_stack)
+
+    def _get_schema_fields(
+        self, cassandra_column_infos: List
+    ) -> Generator[SchemaField, None, None]:
+        # append each schema field (sort so output is consistent)
+        for column_info in cassandra_column_infos:
+            # convert namedtuple to dictionary if it isn't already one
+            column_info = (
+                column_info._asdict()
+                if hasattr(column_info, "_asdict")
+                else column_info
+            )
+            column_info["column_name_bytes"] = None
+            column_name: str = column_info[
+                CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["column_name"]
+            ]
+            cassandra_type: str = column_info[
+                CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["column_type"]
+            ]
+            # if column_info.get("table_name") == "movie_reviews":
+            #     breakpoint()
+            if cassandra_type is not None:
+                self._prefix_name_stack.append(f"[type={cassandra_type}].{column_name}")
+                schema_field_data_type: SchemaFieldDataType = self.get_column_type(
+                    cassandra_type
+                )
+                schema_field: SchemaField = SchemaField(
+                    fieldPath=self._get_cur_field_path(),
+                    nativeDataType=cassandra_type,
+                    type=schema_field_data_type,
+                    description=None,
```

Comment:
Do we get column comment that we can set here ?

Reply:
Cassandra does not support comments on individual columns.